homeassistant/components/velbus/climate.py

```diff
@@ -66,6 +64,14 @@ def current_temperature(self) -> int | None:
         """Return the current temperature."""
         return self._channel.get_state()
 
+    @property
+    def hvac_mode(self) -> HVACMode:
+        """Return the current hvac mode based on cool_mode message."""
+        return {
+            True: HVACMode.COOL,
+            False: HVACMode.HEAT,
+        }.get(self._channel.get_cool_mode())
```

Comment:
Based on a quick glance at the library, `get_cool_mode` may raise an attribute error if we didn't get a `cool_mode` from the device. Either update the library to initialize `_cool_mode` or handle the attribute error here.

Reply:
can you elaborate on this a bit ?

Reply:
This is the implementation in velbusaio: https://github.com/Cereal2nd/velbus-aio/blob/master/velbusaio/channels.py#L551-L552

The problem is that `self._cool_mode` is not guaranteed to be set, and if it isn't, calling get_cool_mode() while cause an attributeerror to be raised.

Reply:
Do I need to make a change in velbusIO then to match: 

    def get_cool_mode(self) -> str:
        if self is none;
           return none
        return self._cool_mode
        
        ?

Reply:
In velbusIO the fix would maybe be like this:
```diff
 class Temperature(Channel):
     """
     A Temperature sensor channel
     """
 
     _cur = 0
     _cur_precision = None
     _max = None
     _min = None
     _target = 0
     _cmode = None
+    _coolmode = None
     _cstatus = None
     _thermostat = False
     _sleep_timer = 0
```